packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -151,32 +151,31 @@ export class DbMutationQueue {
     public lastStreamToken: string,
     // PORTING NOTE: Multi-tab only.
     /**
-     * An identifier for the highest numbered batch that has been acknowledged
-     * or rejected by the server. This ID ensures that even as mutations are
-     * removed, batch IDs never go back in time over the lifetime of a client.
+     * An identifier with the ID of the next mutation batch to be added to this
+     * object store.
      */
-    public lastProcessedBatchId: number
+    public nextBatchId: number
   ) {}
 }
 
 /** keys in the 'mutations' object store are [userId, batchId] pairs. */
 export type DbMutationBatchKey = [string, BatchId];
 
 /**
- * Sets a mutation queue's `lastProcessedBatchId` to its
+ * Sets a mutation queue's `nextBatchId` to the successor of
  * `lastAcknowledgedBatchId` for all existing entries.
  */
-function saveProcessedMutationCount(
+function populateNextMutationBatchId(
   txn: SimpleDbTransaction
 ): PersistencePromise<void> {
-  const mutationsStore = txn.store<DbMutationBatchKey, DbMutationQueue>(
+  const mutationQueuesStore = txn.store<DbMutationQueueKey, DbMutationQueue>(
     DbMutationQueue.store
   );
 
-  let promises = mutationsStore.loadAll().next(entries => {
-    for (const entry of entries) {
-      entry.lastProcessedBatchId = entry.lastAcknowledgedBatchId;
-      promises = promises.next(() => mutationsStore.put(entry));
+  let promises = mutationQueuesStore.loadAll().next(queues => {
+    for (const queue of queues) {
+      queue.nextBatchId = Math.max(queue.lastAcknowledgedBatchId + 1, 1);
```

Comment:
This doesn't seem right.  A client can have lots of unacknowledged mutation batches stored in persistence, right? (e.g. it was operating offline last time the client was used)

But I think it also doesn't matter since IndexedDbMutationQueue.start() is going to replace this with a true globally-unique nextBatchId.  So maybe we don't need this schema migration at all (or we could just write -1 or whatever with a comment explaining why it doesn't matter).